libs/game/sprite.ts

```diff
@@ -214,18 +215,103 @@ class Sprite implements SpriteLike {
     /**
      * Display a speech bubble with the text, for the given time
      * @param text the text to say, eg: ":)"
-     * @param time time to keep text on, eg: 2000
+     * @param time time to keep text on
      */
     //% group="Properties"
     //% blockId=spritesay block="%sprite(agent) say %text||for %millis ms"
-    //% time.defl=2000
+    //% inlineInputMode=inline
     //% help=sprites/sprite/say
-    say(text: string, millis?: number) {
-        this._say = text;
-        if (!millis || millis < 0)
-            this._sayExpires = -1;
-        else
-            this._sayExpires = control.millis() + millis;
+    say(text: string, timeOnScreen?: number, textColor = 15, textBoxColor = 1) {
+        if (!text) {
+            this.updateSay = undefined;
+            return;
+        }
+
+        if (timeOnScreen) {
+            timeOnScreen = timeOnScreen + control.millis();
+        }
+        let pixelsOffset = 0;
+        let holdTextTimer = 1.5;
```

Comment:
I would rename this variable to `holdTextSeconds` or something like that because you are using milliseconds for the other timer